node/overseer/src/lib.rs

```diff
@@ -532,7 +532,7 @@ pub struct Overseer<SupportsParachains> {
 	#[subsystem(blocking, NetworkBridgeRxMessage, sends: [
 		BitfieldDistributionMessage,
 		StatementDistributionMessage,
-		ApprovalDistributionMessage,
+		Box<ApprovalDistributionMessage>,
```

Comment:
Could we just make this an orchestra thing, such that we don’t leak the box at all?

Reply:
Yes, that's what I'm currently working on: an attribute in `orchestra` proc-macro to enable automatic boxing of messages that are more than X bytes size. I haven't mentioned your comment earlier, sorry :)

Reply:
The difficulty is to get a size of some type in compile time though...

Reply:
We can just not automate it yet :) Leave it for later and get some perf numbers for async vs futures instead

Reply:
Yes, it seems that my approaches to do it automatically are quite complicated and ugly. None of them work as I want (partially because Rust have no SFINAE like behaviour apart from autoref concept).

Reply:
Fine tuning what to box and what not sounds like an additional optimization. To get first results, we could just automatically box those subsystem message enums, so the overarching all messages enum stays small. Wouldn't that be enough to get first results?

In other words all variants of the all messages enum contain their data boxed.

```rust
enum AllMessages {
  ApprovalDistributionMessage(Box<ApprovalDistributionMessage>),
  StatementDistributionMessage(Box<StatementDistributionMessage>),
  ...
```

Reply:
Yes, that's the current plan, we have also discussed it yesterday.

Reply:
> The difficulty is to get a size of some type in compile time though...

https://doc.rust-lang.org/stable/std/mem/fn.size_of.html

This is a const fn, so it seems like we should be able to do something with that.